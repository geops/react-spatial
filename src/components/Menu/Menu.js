import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  children: PropTypes.array.isRequired,
  className: PropTypes.string,
  classNameContent: PropTypes.string,
};

const defaultProps = {
  className: 'tm-menu',
  classNameContent: 'tm-menu-content',
};

const Menu = ({ children, className, classNameContent }) => (
  <div className={className}>
    <div className={classNameContent}>{children}</div>
  </div>
);

Menu.propTypes = propTypes;
Menu.defaultProps = defaultProps;

export default Menu;
