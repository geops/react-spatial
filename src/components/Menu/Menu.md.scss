.tm-menu-example {
  position: relative;

  .tm-menu {
    top: 10px;
    left: 10px;
    bottom: 10px;
    position: absolute;
    margin: auto;
    padding: 10px 0;
  }
}
