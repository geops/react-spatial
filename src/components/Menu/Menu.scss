.tm-menu {
  background-color: white;
  border-right: 1px solid #eee;
  border-bottom: 1px solid #eee;
  display: flex;
  flex-direction: column;
  overflow: auto;
  max-height: calc((var(--tm-vh, 1vh) * 100) - 82px);
  max-width: 300px;

  .tm-menu-item-title {
    padding: 0 20px;
    line-height: $btn-size-base;
    font-weight: bold;
    white-space: nowrap;
  }
}

.tm-w-xs {
  .tm-menu {
    max-width: 100vw;
    width: 100vw;
  }
}
