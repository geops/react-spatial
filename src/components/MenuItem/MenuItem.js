import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
  className: PropTypes.string,
  classNameTitle: PropTypes.string,
};

const defaultProps = {
  className: 'tm-menu-item',
  classNameTitle: 'tm-menu-item-title',
};

const MenuItem = ({ children, title, className, classNameTitle }) => (
  <>
    <div className={classNameTitle}>{title}</div>
    <div className={className}>{children}</div>
  </>
);

MenuItem.propTypes = propTypes;
MenuItem.defaultProps = defaultProps;

export default MenuItem;
