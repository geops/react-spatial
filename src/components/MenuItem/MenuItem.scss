.tm-menu .tm-menu-item {
  padding: 0 20px 20px 20px;
}
