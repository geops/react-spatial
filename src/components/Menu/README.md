#

This demonstrates the use of Menu.

```jsx
import React from 'react';
import BasicMap from 'react-spatial/components/BasicMap';
import ConfigReader from 'react-spatial/ConfigReader';
import LayerService from 'react-spatial/LayerService';
import OLMap from 'ol/Map';
import Menu from 'react-spatial/components/Menu';
import MenuItem from 'react-spatial/components/MenuItem';
import ShareMenu from 'react-spatial/components/ShareMenu';
import LayerTree from 'react-spatial/components/LayerTree';

class MenuExample extends React.Component {
  constructor(props) {
    super(props);

    const layerConf = [{
      name: 'OSM Baselayer',
      visible: true,
      data: {
        type: 'xyz',
        url: 'https://c.tile.openstreetmap.org/{z}/{x}/{y}.png',
      },
    }];

    this.map = new OLMap();
    this.layers = ConfigReader.readConfig(this.map, layerConf);
    this.layerService = new LayerService(this.layers);

  }

  render() {
    return (
      <div className="tm-menu-example">
        <BasicMap
          map={this.map}
          zoom={3}
          layers={this.layers}
        />
        <Menu>
          <MenuItem title="Share">
            <ShareMenu url={window.location.href}/>
          </MenuItem>
          <MenuItem title="Layers">
            <LayerTree layerService={this.layerService} />
          </MenuItem>
        </Menu>
      </div>
    );
  }
}

<MenuExample />;
```
