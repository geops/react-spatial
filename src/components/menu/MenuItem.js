import React from 'react';
import PropTypes from 'prop-types';

import './MenuItem.scss';

const propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
};

const MenuItem = ({ children, title }) => (
  <>
    <div className="tm-menu-item-title">{title}</div>
    <div className="tm-menu-item">{children}</div>
  </>
);

MenuItem.propTypes = propTypes;

export default MenuItem;
