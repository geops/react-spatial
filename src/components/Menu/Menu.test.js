import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Menu from './Menu';
import MenuItem from '../MenuItem';

configure({ adapter: new Adapter() });

describe('Menu', () => {
  test('should match snapshot.', () => {
    const component = renderer.create(
      <Menu>
        <MenuItem title="foo">
          <div>foo</div>
        </MenuItem>
        <MenuItem title="bar">
          <div>bar</div>
        </MenuItem>
      </Menu>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
