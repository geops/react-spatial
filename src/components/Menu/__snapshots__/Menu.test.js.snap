// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match snapshot. 1`] = `
<div
  className="tm-menu"
>
  <div
    className="tm-menu-content"
  >
    <div
      className="tm-menu-item-title"
    >
      foo
    </div>
    <div
      className="tm-menu-item"
    >
      <div>
        foo
      </div>
    </div>
    <div
      className="tm-menu-item-title"
    >
      bar
    </div>
    <div
      className="tm-menu-item"
    >
      <div>
        bar
      </div>
    </div>
  </div>
</div>
`;
